channeld/channeld.c

```diff
@@ -1602,12 +1601,16 @@ static void handle_peer_fail_malformed_htlc(struct peer *peer, const u8 *msg)
 	 *    `update_fail_malformed_htlc`:
 	 *      - MUST fail the channel.
 	 */
-	if (!(failure_code & BADONION)) {
+	/* We only handle these cases. */
```

Comment:
OK, I converted unknown ones to temporary_failure.

There are two channelds involved here.  This path is the outgoing one: it sets failcode to one of these three values.  The incoming channeld turns that failcode into a message.  It only knows how to handle these three types: for any new type it would hit this code in make_failmsg:

	status_failed(STATUS_FAIL_INTERNAL_ERROR,
		      "Asked to create failmsg %u (%s)",
		      failcode, onion_type_name(failcode));

which would cause failure of the *incoming* channel.